vpype_gcode/gwrite.py

```diff
@@ -18,10 +18,20 @@ def invert_axis(document: vp.Document, invert_x: bool, invert_y: bool):
     """Inverts none, one or both axis of the document.
     This applies a relative scale operation with factors of 1 or -1
     on the two axis to all layers. The inversion happens relative to
+    the center of the page, if the page size has been set, otherwise to
     the center of the bounds.
     """
 
-    bounds = document.bounds()
+    # if the page size is set, then we'll flip around the middle of the page,
+    # otherwise around the geometry bounds
+    if document.page_size is not None:
+        # page size is in px, but everything else is in mm
+        (x, y) = document.page_size
+        px_per_mm = vp.utils.UNITS["mm"]
```

Comment:
`vp.UNITS` would be the canonical syntax here (vpype imports everything public in the top-level namespace).

Reply:
Also, this is very likely incorrect. At this stage, the document is converted to whatever unit is set by the config file (see [this line](https://github.com/plottertools/vpype-gcode/blob/595b15bd2fbb90b89349b1cc299875d206d79be3/vpype_gcode/gwrite.py#L115). Hard-coding units here will likely break any setup with a different unit. Some refactoring might be needed to address this.

Reply:
Ahh, so that is what's going on! Was wondering about things being in mm. Seemed strange.